pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,284 @@
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=test-cluster-north \
+				   --leader-clusterset=test-cluster-set \
+				   --secret=member-access-token \
+				   --leader-server=https://172.18.0.3:6443 \
+				   --clusterset=test-clusterset \
+				   --namespace=kube-system \
+				   --clusterid=test-cluster-east
+
+# Join the ClusterSet from a member cluster with a config file. If both command line options and config file are specified, 
+the arguments in config file will be used.
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetConfig
+memberCluster:
+  namespace: kube-system
+  clusterid: test-cluster-east
+  clusterSet: test-clusterset
+leaderCluster:
+  namespace: antrea-multicluster
+  secret: member-access-token
+  APIServer: https://172.18.0.3:6443
+  clusterid: test-cluster-north
+  clusterSet: test-clusterset
+`, "\n")
+
+func NewJoinCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := joinOptions{}
+	joinOpts = &o
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Namespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.ClusterID, "leader-clusterid", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Secret, "secret", "", "", "Secret of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.ClusterSetID, "leader-clusterset", "", "", "ClusterSet of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.APIServer, "leader-server", "", "", "APIServer endpoint of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.ClusterSetID, "clusterset", "", "", "ClusterSet of the member cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.Namespace, "namespace", "", "", "Namespace of the member cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.ClusterID, "clusterid", "", "", "Cluster ID of the member cluster")
+	command.Flags().StringVarP(&joinOpts.ConfigFile, "config-file", "f", "", "Config file that defines all config options")
+
+	return command
+}
+
+func joinRunE(cmd *cobra.Command, args []string) error {
+	var err error
+	if err = joinOpts.validateAndComplete(); err != nil {
+		return err
+	}
+	k8sClient, err := common.NewClient(cmd)
+
+	memberClusterNamespace := joinOpts.MemberClusterOpt.Namespace
+	memberClusterID := joinOpts.MemberClusterOpt.ClusterID
+	memberClusterSet := joinOpts.MemberClusterOpt.ClusterSetID
+	createdRes := []*unstructured.Unstructured{}
+	defer func() {
+		if err != nil {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to join the ClusterSet. Deleting the created resources.\n")
```

Comment:
Not all your defer functions have this kind of warning log, could you check to remove or add for all cases to make then consistent?

Reply:
Add warning log to others. Done.